contact.php

```diff
@@ -244,14 +244,15 @@
 			);
 			$db->insert_query("maillogs", $log_entry);
 		}
-
-		if($mybb->usergroup['emailfloodtime'] > 0 || (isset($sent_count) && $sent_count + 1 >= $mybb->usergroup['maxemails']))
+		
+		$mybb->input['from'] = $mybb->get_input('from');
+		if(isset($mybb->input['from']) && !empty($mybb->input['from']))
```

Comment:
This can be simplified:

```suggestion
		if(!empty($mybb->input['from']))
```